service/frontend/workflow_handler.go

```diff
@@ -3036,6 +3040,10 @@ func (wh *WorkflowHandler) DescribeSchedule(ctx context.Context, request *workfl
 
 		// Search attributes in the Action are already in external ("aliased") form. Do not alias them here.
 
+		// proceed only if the schedule workflow is still running
```

Comment:
you are right, this should have been after the for loop, we just don't want to send a refresh signal to the schedule workflow, otherwise this fails with an error that workflow is already completed. 

We are not waiting for all running workflows to complete before completing the schedule workflow. are there any side effects if we don't wait for those workflows to complete? 

Also, I think we are already remove the completed workflows from the results here since we have the following in a few lines below this: 
```
for _, ex := range queryResponse.GetInfo().GetRunningWorkflows()
``` 

Or are you referring to sth else?

Reply:
The API behavior that I want is: we should return an accurate list of running workflows as of some instant in time during the processing of the request. I.e. if one or more of the "running" workflows that we get from querying the schedule workflow have completed already, then we should not return them to the user. And same for the close result.

Now the question is, how to accomplish this. The current implementation does an admittedly-convoluted thing: if it sees "running" workflows from the query that are not running anymore, it signals the schedule workflow to refresh its view of running workflows, then queries again.

If we leave running workflows when we close, then the close result will contain them forever, until something restarts the schedule workflow. But it seems silly to restart it just to notice that they're closed and then close itself again. The cheaper option is to just remove them from the list that gets returned and not touch the workflow.

But actually just removing them from the list works all the time. I.e. we could actually skip the whole refresh/query again thing and treat running and closed cases the same. The only downside is that that for repeated calls to DescribeSchedule, we'll do those GetMutableState calls every time, instead of having the schedule workflow fix its state to drop them.

Reply:
Thanks for the explanation, it makes sense. I just pushed a commit to fix this, please take a look and let me know what you think.